### PR TITLE
US20513 Shows Landing Page

### PR DESCRIPTION
**Pull request by kellgauthier**
## Task
Develops shows landing page per [design](https://projects.invisionapp.com/d/main#/console/20964701/441631718/preview). 
[Rally](https://rally1.rallydev.com/#/66096747656ud/custom/24109743995?detail=%2Fuserstory%2F483183596816) 
**[Preview](https://deploy-preview-1067--crds-mediaint.netlify.app/shows/)**

## Considerations 
- Jumbotron is a static image, will need to update with video background (May need to do a contentful-migration to add a field to a collection for a trailer) 
- Data in int does not match design 
- Unsure where carousel images and recently added images are supposed to link to 
- Should content administrators be able to eventually choose what shows go into the carousel?

### CODE REVIEW

File: _assets/stylesheets/pages/_shows.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
+.shows-landing-jumbotron {
+  .jumbotron-content {
+    p { 
+      width: 40%;
```
Comment (silasbaur) [inline _assets/stylesheets/pages/_shows.scss:4]: Not a fan of this. Lets just add a class of `col-sm-4` to the div with `jumbotron-content`.

File: _assets/stylesheets/pages/_shows.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+  }
+}
+
+.feature-logo {
+  max-width: 245px;
+}
+
+.recently-added {
+  margin-bottom: -9rem;
```
Comment (silasbaur) [inline _assets/stylesheets/pages/_shows.scss:14]: We do not need this. Change the layout to `default_footer_flush`

File: _plugins/generators/shows_generator.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module ShowsGenerator
+  class Generator < Jekyll::Generator
+    def generate(site)
+      site.data['shows'] = []
+      videos = site.collections['videos'].docs
+      videos.each do |video|
+        shows = site.collections['videos'].docs.select{ |doc| doc.data.dig('collections') }
+        if shows.present?
+          testing = video.data['shows_episodes'] = site.collections['videos'].docs.select{ |doc| doc.data.dig('collections') }.flatten.uniq
```
Comment (silasbaur) [inline _plugins/generators/shows_generator.rb]: We do not need `testing =` anymore. If we do need it we should not call this testing

File: _includes/_shows-most-recent.html [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<div class="mobile-soft-left">
+  <img class="recently-added-image" src="{{ episode.image.url | imgix: site.imgix }}">
+  <h2 class="shows-landing-page-subtitle text-uppercase flush-top text-orange">{{ episode.title }}</h2>
+  <h3 class="font-family-condensed-extra text-uppercase text-white flush-top shows-recently-added-subtitle">{{ episode.title }}</h3>
+  <p class="font-size-smaller soft-half-top text-white shows-related-description">
+    {{ episode.description }}
+  </p>
+</div>
```
Comment (silasbaur) [inline _includes/_shows-most-recent.html:8]: (PICKY COMMENT INCOMING) Not a huge fan of how this set up. I think it is more readable if it is set up like this:
```
<div>
  <div>
    <img />
  </div>

  <div>
    {{ Content here }}
  </div>
</div>
```

File: shows.html [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    <div class="bg-video-player">
+      <video
+        width="1800"
+        height="1013"
+        class="bg-jumbotron-video hide"
+        autoplay="autoplay"
+        loop="loop"
+        muted
+        playsinline
+        src="https://videos.ctfassets.net/y3a9myzsdjan/61UX4j16LJ4i6Hul1obi2Z/02c4498b9d0c1ef87fb7767fbfcaed54/spiritual-outfitters-loop.mp4"
```
Comment (silasbaur) [inline shows.html:21]: Why is this here? Shouldn't this be a featured show episode?

File: shows.html [hunk 6]
```diff
@@ -0,0 +1,86 @@
+---
+layout: default
+snail_trail: disabled
+---
+{% assign featured_shows = site.collections | where: 'featured_on_shows_landing_page', true %}
+{% assign recent_episodes = site.data['shows'] %}
+
+<section class="shows-landing-jumbotron">
+  <div class="jumbotron jumbotron-xl bg-video inline-video jumbotron-home flush" style="text-align: left; background-image: url('//crds-media.imgix.net/4b1Vka2whxFu1JNjx9pckB/1d48c4bbc5399c597c39a81fc625e94e/Screen_Shot_2021-01-05_at_2.35.55_PM.png?{{ site.imgix_params.placeholder_sixteen_nine }}');" data-optimize-bg-img >
```
Comment (silasbaur) [inline shows.html:9]: We should probably create a featured_media entry for this so that can be easily changed by content admins

File: shows.html [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+          <div class="jumbotron-content mobile-soft-left">
+            {% for shows in featured_shows offset: 1  limit: 1 %}
+            <img alt="Garage Show Series" class="feature-logo" src="{{shows.logo.url | imgix: site.imgix }}">
+            <h3 class="font-family-condensed-extra text-uppercase text-white shows-landing-page-title">Episode 24: Why do we have a garage?</h3>
+            <p class="font-size-smaller soft-half-top text-white">
+              {{ shows.description }}
+            </p>
+            {% endfor %}
+            <div class="shows-button-container soft-half-top">
+              <crds-button href="#" color="white" icon="play-thin" icon-align="left" icon-color="blue" icon-size="14" text="Play"></crds-button>
```
Comment (silasbaur) [inline shows.html:36]: This button needs to be able to play the video

**Review by silasbaur — CHANGES_REQUESTED** [review]
Let me know if you want to talk through anything
1. Everything is wrapped in `col-sm-10 col-sm-offset-1` but the design does not look like it calls for that
2. Remove arrows on the carousel if there are not more than 4 shows
3. Image does not line up
Screen Shot 2021-02-03 at 10 27 25 AM
4. Take a look at mobile

**Review by silasbaur — CHANGES_REQUESTED** [review]
1. Everything is wrapped in `col-sm-10 col-sm-offset-1` but the design does not look like it calls for that
2. Image does not line up
Screen Shot 2021-02-03 at 10 27 25 AM
3. Take a look at mobile.

File: _assets/stylesheets/pages/_shows.scss [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+.shows-recently-added-subtitle {
+  font-size: 27px;
+  font-weight: 100 !important;
+}
+
+.shows-related-description {
+  display: flex;
+}
+
+video {
```
Comment (silasbaur) [inline _assets/stylesheets/pages/_shows.scss:73]: Use a class

**Review by silasbaur — CHANGES_REQUESTED** [review]
1. Inconsistant image sizes
Screen Shot 2021-02-10 at 9 58 33 AM
2. The play icon disappear after it is clicked. Is there a paused icon that can be used for when the button says pause?

**Comment by kellgauthier** [comment]
> 1. Inconsistant image sizes
>    Screen Shot 2021-02-10 at 9 58 33 AM
> 2. The play icon disappear after it is clicked. Is there a paused icon that can be used for when the button says pause?

**Review by silasbaur — APPROVED** [review]

**Review by silasbaur — CHANGES_REQUESTED** [review]
Screen Shot 2021-02-10 at 11 10 38 AM

**Review by silasbaur — APPROVED** [review]

**Pull request merged**